### ISSUE TITLE
Metabase ingress errors with extensions/v1beta1 in 1.23 cluster

### ISSUE THREAD

**Issue by Firaenix** [pmint93/helm-charts#32]
Hi,

I'd love to use this helm chart however I'm getting the following error when I deploy to my Kubernetes 1.23 cluster.

`Error: unable to recognize "": no matches for kind "Ingress" in version "extensions/v1beta1"`

Is it possible that `Capabilities.KubeVersion.GitVersion` in ingress.yaml should be replaced with `Capabilities.KubeVersion.Version` as shown here? https://helm.sh/docs/chart_template_guide/builtin_objects/

Thanks

**Comment by pmint93** [comment]
Changes in helm cause `Capabilities.KubeVersion.GitVersion` to be replaced by `Capabilities.KubeVersion.Version`

Found the commit here: https://github.com/helm/helm/commit/097834de0ad7e33fc5f13088fbb5a17128fb21e3#diff-a15c00615f24e3d585c1e7dee030d66e0e1d1666a34ce1ff038a84dbe2f5dba5R104

It seem the commit has been released since helm 3. But I didn't see any mention in helm release change log.
Moreover, at the time the version compare was added to this chart in #16, I see @lgg42 also using helm 3. So, why he didn't get the error ?

Hmm, need more investigation

**Comment by Firaenix** [comment]
Hmm, I'm honestly not sure which version of Helm I'm using as I'm using Helm via terraform with this module: https://github.com/hashicorp/terraform-provider-helm. its definitely Helm 3 though.

I dont know if this is related but I'm using microk8s as my k8s distro and my version of all of my nodes is v1.23.3-2+d441060727c463. maybe the version isnt being parsed correctly?

**Comment by pmint93** [comment]
Yeah definitely MicroK8s related, see helm/helm#9371

MicroK8s version isn't SemVer compliant, then helm couldn't correctly compare them. I think we should find a way to workaround this

**Comment by Firaenix** [comment]
Is it possible to have a flag for v1beta vs v1 in values.yaml?

**Comment by pmint93** [comment]
@Firaenix  Nah I don't think so. Instead, we can reverse the if condition so default it use the `networking.k8s.io/v1`

eg.

```
{{- if semverCompare "<1.19" .Capabilities.KubeVersion.GitVersion -}}
apiVersion: extensions/v1beta1
...
{{- else -}}
apiVersion: networking.k8s.io/v1
...
```
And, it reasonable while user eventually upgrade their clusters to newer versions

**Comment by Firaenix** [comment]
sounds great to me

**Comment by pmint93** [comment]
Fixed in #36
